File: 04-vector-search/python/SearchComponents/indexes.py
def create_vector_indexes(collection, index_list, db, collection_name):

    collection_indexes = collection.index_information()    

    for indexname, vectorColumn in index_list:
        for index in collection_indexes:
            if index == indexname:
                collection.drop_index(indexname)
                break
        
        db.command({
            'createIndexes': collection_name,
            'indexes': [
                {
                'name': indexname,
                'key': {
                    f"{vectorColumn}": "cosmosSearch"
                },
                'cosmosSearchOptions': {
                    'kind': 'vector-ivf',
                    'numLists': 1,
                    'similarity': 'COS',
                    'dimensions': 1536
                }
                }
            ]
            })

File: 04-vector-search/node.js/SearchComponents/indexes.js
async function createVectorIndexes(collection, indexList, db, collectionName) {
    const collectionIndexes = await collection.indexInformation();

    for (let [indexName, vectorColumn] of indexList) {
        for (let index of Object.keys(collectionIndexes)) {
            if (index === indexName) {
                await collection.dropIndex(indexName);
                break;
            }
        }

        const commandResult = await db.command({
            'createIndexes': collectionName,
            'indexes': [
                {
                    'name': indexName,
                    'key': {
                        [vectorColumn]: "cosmosSearch"
                    },
                    'cosmosSearchOptions': {
                        'kind': 'vector-ivf',
                        'numLists': 1,
                        'similarity': 'COS',
                        'dimensions': 1536
                    }
                }
            ]
        });
    }
}